pallets/randomness/src/traits.rs

```diff
@@ -21,12 +21,13 @@ pub trait SendRandomness<AccountId, R> {
 	fn send_randomness(contract: AccountId, randomness: R);
 }
 
-/// Get the epoch number
-/// Optimize design to only update storage values when they change
-pub trait GetEpochNumber<Index> {
-	// get the last epoch number
-	fn get_epoch_number() -> Index;
-	// epoch changed in this relay chain block
+/// Get the epoch index
+pub trait GetEpochIndex<Index> {
```

Comment:
Is this a well known key we can fetch from the relay?

Reply:
Or is it present in Parachain System?

Reply:
it is not, I am going to add it well known keys but we can hardcode the key locally until we get it in upstream.

the blocker for it is the ability to read arbitrary entries in the proof, that is currently private but I made a PR to change it (and the patch on our most recent dep update so we can use it before it is merged and brought in)

Reply:
added to well-known keys https://github.com/paritytech/polkadot/pull/5259

Reply:
awesome, did we cherry-pick it in our branch? or will we wait for 0.9.19?